src/codegen.cpp

```diff
@@ -1897,6 +1897,17 @@ static void coverageAllocLine(StringRef filename, int line)
     allocLine(coverageData[filename], line);
 }
 
+extern "C" JL_DLLEXPORT void jl_coverage_visit_line(const char* filename, size_t len_filename, int line)
+{
+    assert(!imaging_mode);
+    if (filename == "" || filename == "none" || filename == "no file" || filename == "<missing>" || line < 0)
```

Comment:
these comparisons are always false.
```suggestion
    if (filename == StringRef("") || filename == StringRef("none") || filename == StringRef("no file") || filename == StringRef("<missing>") || line < 0)
```